src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +32,10 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i = i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .filter(i -> i % 2 != 0)
+                .average().orElseThrow(NoSuchElementException::new);
```

Comment:
```suggestion
                .average()
                .orElseThrow();
```

What exception methos orElseThrow throws by default?